lib/projectPreprocessor.js

```diff
@@ -163,7 +163,9 @@ class ProjectPreprocessor {
 		if (processedProject) {
 			if (processedProject.ignored) {
 				log.verbose(`Dependency of project ${parent.id}, "${project.id}" is flagged as ignored.`);
-				parent.dependencies.splice(parent.dependencies.indexOf(project), 1);
+				if (parent.dependencies.indexOf(project) !== -1) {
```

Comment:
You might use Array#includes instead
```suggestion
				if (parent.dependencies.includes(project)) {
```